source/ada/lsp-ada_handlers.adb

```diff
@@ -1318,7 +1318,7 @@ package body LSP.Ada_Handlers is
             if Other_Part = No_Defining_Name then
                --  No next part is found. Check first defining name
                Other_Part := Laltools.Common.Find_Canonical_Part
-                 (Definition, Self.Trace);
+                 (Definition, Self.Trace, Imprecise_Fallback => True);
```

Comment:
Same remark as above: add a 2-step mechanism to make sure we set `Imprecise` when `Imprecise_Fallback` is needed.